Add post_process_depth_estimation to image processors and support ZoeDepth's inference intricacies (#32550)
huggingface/transformers: docs/source/en/tasks/monocular_depth_estimation.md

>>> with torch.no_grad():
...     outputs = model(pixel_values)
...     outputs_flip = model(pixel_values=torch.flip(inputs.pixel_values, dims=[3]))
```

<Tip>

> According to the [original implementation](https://github.com/isl-org/ZoeDepth/blob/edb6daf45458569e24f50250ef1ed08c015f17a7/zoedepth/models/depth_model.py#L131) ZoeDepth model performs inference on both the original and flipped images by default and averages out the results. The `post_process_depth_estimation` function will handle this for us by passing the flipped outputs to the optional `outputs_flip` argument.

Reviewer: ```suggestion
In the [original implementation](https://github.com/isl-org/ZoeDepth/blob/edb6daf45458569e24f50250ef1ed08c015f17a7/zoedepth/models/depth_model.py#L131) ZoeDepth model performs inference on both the original and flipped images and averages out the results. The `post_process_depth_estimation` function can handle this for us by passing the flipped outputs to the optional `outputs_flip` argument.
```
